### ISSUE TITLE
Retrain Inception Model Error In Android

### ISSUE THREAD

**Issue by kinDSa** [tensorflow/tensorflow#8396]
I am following [Tensorflow for poet](https://codelabs.developers.google.com/codelabs/tensorflow-for-poets/#0) instruction for retrain model. I have successfully create retrained_graph.pb and retrained_labels.txt. While I use imagenet_comp_graph_label_strings.txt and tensorflow_inception_graph.pb then application run without any error. But use my created file then I get error that:

`Caused by: java.lang.UnsupportedOperationException: Op BatchNormWithGlobalNormalization is not available in GraphDef version 21. It has been removed in version 9. Use tf.nn.batch_normalization().
                                                                         at org.tensorflow.Graph.importGraphDef(Native Method)
                                                                         at org.tensorflow.Graph.importGraphDef(Graph.java:118)
                                                                         at org.tensorflow.Graph.importGraphDef(Graph.java:102)
                                                                         at org.tensorflow.contrib.android.TensorFlowInferenceInterface.load(TensorFlowInferenceInterface.java:402)
                                                                         at org.tensorflow.contrib.android.TensorFlowInferenceInterface.initializeTensorFlow(TensorFlowInferenceInterface.java:91)
                                                                     	at org.tensorflow.demo.TensorFlowImageClass`

**Comment by drpngx** [comment]
As the error message says, you want to use `tf.nn.batch_normalization` instead. You may have to retrain from scratch.

@nealwu may have some suggestion.

**Comment by choonguri** [comment]
I used [this](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/examples/image_retraining)(image retraining sample - inception v3) to learn my own classification.
When I used my 'pb' and 'txt' files in Android, I got the same error.

**Comment by asimshankar** [comment]
Thanks for the bug report, I believe there is an unexpected mismatch in how we import graphs in Python and the C API, looking into it.

**Comment by kinDSa** [comment]
@drpngx, As per your suggestion I try to retrain model from scratch followed [TensorFlow guides ](https://www.tensorflow.org/versions/r0.11/how_tos/image_retraining/) but I get same error.
I also try to strip graph and change in [ClassifierActivity](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/android/src/org/tensorflow/demo/ClassifierActivity.java) Activity. But still I get same error.

**Comment by kinDSa** [comment]
I get one blog [Tensorflow for mobile](https://petewarden.com/2016/09/27/tensorflow-for-mobile-poets/).Using That create **optimized_graph.pb, rounded_graph.pb, mmapped_graph.pb**  files. 

optimized_graph.pb and rounded_graph.pb file work in android application without any error.
While use mmapped_graph.pb I get error that  **Failed to initialize: java.io.IOException: Not a valid TensorFlow Graph serialization: Invalid GraphDef**

Performance of application is not good while use optimized_graph.pb and rounded_graph.pb file.While application camera screen not contain any flower photos otherwise random flower name show with high confidence rate. Any reason behind these. 
![screenshot](https://cloud.githubusercontent.com/assets/25680329/24043258/cad02cb8-0b3b-11e7-96d4-11ff5b34f0c9.png)
